src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2040 @@
+        self.queue_command_responses = Queue.Queue(maxsize=0)
+        # save messages (error_code = 4,5) from commandport
+        self.queue_img = Queue.Queue(maxsize=0)
+
+        self.should_listen = True  # used in readCommandResponse thread
+        self._waitForCorrectResponse = True  # used in sendCommand
+        self._getReadoutCamInfo = False  # nasty trick to get cam info
+
+        # start thread, which keeps reading the commandport response continuously
+        self._start_receiverThread()
+
+        # Note: start HPDTA after initializing queue and command and receiver treads
+        # but before image thread and initializing children!
+
+        # TODO check if already running....otherwise start multiple apps
+        # TODO  -> in acquisition mode it looks like it does not start a second app, but also does not report that
+        # TODO -> in processing mode it is possible to start multiple apps....
+        # TODO find out where to ask for acq or processing mode
+        # TODO is there a clever way for checking if app still running? Seems to be no command available to check
+        # TODO appEnd only works for the last opened window
+        # TODO want to check if we want to start app invisible (sVisible = False)
+
+        # self.AppStart() # start HPDTA software  # TODO for testing in order to not start a new App
+
+        if children:
+            try:
+                kwargs = children["readoutcam"]
+            except Exception:
+                raise
+            self._readoutcam = ReadoutCamera(parent=self, spectrograph=children.get("spectrograph"),
+                                             daemon=daemon, **kwargs)
+            self.children.value.add(self._readoutcam)  # add readoutcam to children-VA
+            try:
+                kwargs = children["streakunit"]
+            except Exception:
+                raise
+            self._streakunit = StreakUnit(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._streakunit)  # add streakunit to children-VA
+            try:
+                kwargs = children["delaybox"]
+            except Exception:
+                raise
+            self._delaybox = DelayGenerator(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._delaybox)  # add delaybox to children-VA
+
+    def _openConnection(self):
+        """
+        open connection with RemoteEx client.
+        :parameter host: IP-adress or hostname
+        :parameter port: port for sending/receiving commands
+        :parameter port_d: port for reading images
+        return: connection to RemoteEx command and data port
+        """
+        # connect to sockets
+        try:
+            self._commandport = socket.create_connection((self.host, self.port), timeout=5)
+            self._dataport = socket.create_connection((self.host, self.port_d), timeout=5)
+        except socket.timeout as msg:
```

Comment:
HwError is not properly created. It should just be `HwError("Failed....")`.
Also, if the two exceptions indicate slightly different issues (example: no connection to the computer vs RemoteEx is not running), make it clear in the error messages. Otherwise, merge the two types of exceptions into one block.

Also, all the "...." in the exception messages must be entirely written. If they are all actually the same message, you could just define the message in a constant at the top of the function, and reuse it.

Reply:
What is exactly the difference between a socket.timeout (just no response??) and a socket.error (??)